backend/src/main/scala/bloop/Compiler.scala

```diff
@@ -702,13 +722,20 @@ object Compiler {
     val newStamps = {
       import sbt.internal.inc.Stamps
       val oldStamps = analysis.stamps
-      val oldProducts = oldStamps.products.map {
+      // Use empty stamps for files that have fatal warnings so that next compile recompiles them
+      val rebasedSources = oldStamps.sources.map {
+        case t @ (file, _) =>
+          // Assumes file in reported diagnostic matches path in here
+          if (!sourceFilesWithFatalWarnings.contains(file)) t
```

Comment:
maybe reversing the condition to make it easier to read?

Reply:
Will do the same as in the other suggestion :wink: